pkg/framework/plugins/nodeutilization/nodeutilization.go

```diff
@@ -73,6 +96,68 @@ func normalizePercentage(percent api.Percentage) api.Percentage {
 	return percent
 }
 
+func getNodeThresholdsFromAverageNodeUsage(
+	nodes []*v1.Node,
+	usageClient usageClient,
+	lowSpan, highSpan api.ResourceThresholds,
+) map[string][]api.ResourceThresholds {
+	total := api.ResourceThresholds{}
+	average := api.ResourceThresholds{}
+	numberOfNodes := len(nodes)
+	for _, node := range nodes {
+		usage := usageClient.nodeUtilization(node.Name)
+		nodeCapacity := node.Status.Capacity
+		if len(node.Status.Allocatable) > 0 {
+			nodeCapacity = node.Status.Allocatable
+		}
+		for resource, value := range usage {
+			nodeCapacityValue := nodeCapacity[resource]
+			if resource == v1.ResourceCPU {
+				total[resource] += api.Percentage(value.MilliValue()) / api.Percentage(nodeCapacityValue.MilliValue()) * 100.0
+			} else {
+				total[resource] += api.Percentage(value.Value()) / api.Percentage(nodeCapacityValue.Value()) * 100.0
+			}
+		}
+	}
+	lowThreshold, highThreshold := api.ResourceThresholds{}, api.ResourceThresholds{}
+	for resource, value := range total {
+		average[resource] = value / api.Percentage(numberOfNodes)
+		// If either of the spans are 0, ignore the resource. I.e. 0%:5% is invalid.
+		// Any zero span signifies a resource is either not set or is to be ignored.
+		if lowSpan[resource] == MinResourcePercentage || highSpan[resource] == MinResourcePercentage {
+			lowThreshold[resource] = 1
+			highThreshold[resource] = 1
+		} else {
+			lowThreshold[resource] = normalizePercentage(average[resource] - lowSpan[resource])
+			highThreshold[resource] = normalizePercentage(average[resource] + highSpan[resource])
+		}
+	}
+
+	nodeThresholds := make(map[string][]api.ResourceThresholds)
+	for _, node := range nodes {
+		nodeThresholds[node.Name] = []api.ResourceThresholds{
+			lowThreshold,
+			highThreshold,
+		}
+	}
+	return nodeThresholds
+}
+
+func getStaticNodeThresholds(
+	nodes []*v1.Node,
+	usageClient usageClient,
```

Comment:
This `usageClient` isn't used by the function.

Reply:
Removed